### PR TITLE
Migrate log agent component to the new file hierarchy

### PR DESCRIPTION
**Pull request by ogaca-dd**
### What does this PR do?

This PR migrates the log agent component to the new file hierarchy where the `Component` interface is not in the same package as the implementation.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: comp/logs/agent/agentimpl/agent.go [hunk 1]
```diff
@@ -3,7 +3,8 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2023-present Datadog, Inc.
 
-package agent
+// Package agentimpl contains the implementation of the logs agent component.
+package agentimpl
```
Comment (hush-hush) [inline comp/logs/agent/agentimpl/agent.go:7]: Do we have multiple implementation for `comp/logs/agent` ? If not this folder should be named `impl`.

**Review by hush-hush — APPROVED** [review]

**Review by gh123man — APPROVED** [review]

**Review by hghotra — APPROVED** [review]
LGTM - Serverless

**Review by lebauce — APPROVED** [review]

**Comment by GustavoCaso** [comment]
Great work 🎉 

I believe we can update the list of component that needs to be migrated in the `task/components.py` file https://github.com/DataDog/datadog-agent/blob/7d1b66032bca20706e81083100b6d60cc12ef789/tasks/components.py#L55

**Pull request merged**